prolif/residue.py

```diff
@@ -221,7 +231,7 @@ def __getitem__(self, key):
             f"Expected a ResidueId, int, or str, got {type(key).__name__!r} instead",
         )
 
-    def select(self, mask):
+    def select(self, mask: NDArray[np.bool_]) -> "ResidueGroup":
```

Comment:
On this one it's fine to just use `Any` tbh, even if the docstring is a bit more specific, it seems to fix the issue introduced by typing `_residues` as a `Sequence`, and technically we can use more than `NDArray[bool]` to index/slice another `NDArray`